src/api-routes/books/{id}.js

```diff
@@ -22,4 +22,34 @@ const get = async (req, res) => {
   }
 };
 
-export { get };
+/**
+ * Update book by unique ID
+ *
+ * @type {RequestHandler}
+ */
+const patch = async (req, res) => {
+  try {
+    const { id } = req.params;
+
+    // Fetch the existing book
+    const existingBook = await getBookById(id);
+
+    if (!existingBook) {
+      errorBuilder(res, 404, 'A book with the specified ID was not found.');
+    } else {
+      const updateData = req.body.data.attributes;
+      const updatedBook = await updateBookById(id, updateData, existingBook);
+
+      if (!updatedBook) {
+        errorBuilder(res, 500, 'Failed to update the book.');
```

Comment:
Hmm, under what circumstances, this error will be returned from the db? Does it mean that nothing has been changed on the db side? If so, why this should be returned as a 500 (internal server error)?

Reply:
I believe if anything else happened other than 404 on a resource it will give 500 as an error. I may think more specific to describe this error. But this is what I thought of.

Reply:
done.